### PR TITLE
DC-354(Bug): Fallback for claim names related to IAL level and ID Proofing

### PR DESCRIPTION
**Pull request by noxmwalsh**
#### 🔗 Jira ticket
[DC-354](https://codeforamerica.atlassian.net/browse/DC-354)

#### ✍️ Description

This PR extends OIDC ID-proofing claim handling so Colorado myColorado can supply verification level and date via dedicated claims when Socure referencing claims are unavailable.  The most common scenario where this happens is if the user ID proofs via the MyCO mobile app.  

**Behavior**

- **Verification level:** After evaluating `Oidc:VerificationClaims:LevelClaimName` (default `socureIdVerificationLevel`), the translator falls back to a secondary claim when the primary is missing, empty, or not a recognized numeric level (`1`, `1.5`, `2`).
- **Verification date:** After evaluating `Oidc:VerificationClaims:DateClaimName` (default `socureIdVerificationDate`), the translator falls back when the primary is missing or not parseable as a date.
- **Configurable fallbacks:** `FallbackLevelClaimName` and `FallbackDateClaimName` are optional settings under `Oidc:VerificationClaims`. When omitted or whitespace, defaults match myColorado’s `MyCoIdVerificationLevel` and `myCoIdVerificationDate`.
- Primary claims still win when both primary and fallback are present and valid.

**Files touched**

- `OidcVerificationClaimSettings` - added optional `FallbackLevelClaimName` / `FallbackDateClaimName`.
- `OidcVerificationClaimTranslator` - resolved primary/fallback chain for level and date; small fix for nullable dictionary values in `TryGetNonEmptyClaimValue`.
- `appsettings.co.example.json` (CO) - documents all four verification claim keys for copy/paste and local parity with deployed config.
- `tofu/config/dev-co/main.tf` - sets `Oidc__VerificationClaims__FallbackLevelClaimName` and `Oidc__VerificationClaims__FallbackDateClaimName` alongside existing Level/Date env vars.
- Unit tests — coverage for Socure-only, MyCO-only, primary-vs-fallback precedence, custom fallback names via settings, and date edge cases.

#### 🔗 Links to related PRs
N/A

#### ✅ Completion tasks

- [x] Added relevant tests
- [ ] Meets acceptance criteria in ticket
- [ ] Configuration changes:
  - [x] If new environment variables are added, update in Tofu *(fallback claim names added to `dev-co/main.tf`; not secrets)*
  - [x] If new environment secrets are added, update in Tofu and set in AWS Secret Manager *(N/A — no new secrets)* (Added in `main.tf` in dev-co.tf)
  - [x] If you're adding an appsetting, add it to the requisite example file, and update it in the AWS AppConfig *(example file updated)
  - [ ] If appsettings are changed, update in AWS AppConfig**

[DC-354]: https://codeforamerica.atlassian.net/browse/DC-354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/SEBT.Portal.Infrastructure/Services/OidcVerificationClaimTranslator.cs [hunk 1]
```diff
@@ -52,6 +62,63 @@ public OidcVerificationClaimTranslator(
...
+
+    private static bool ClaimNamesAreEquivalent(string a, string b) =>
+        string.Equals(a, b, StringComparison.OrdinalIgnoreCase);
+
+    private static bool TryGetNonEmptyClaimValue(
+        IReadOnlyDictionary<string, string> claims,
+        string claimName,
+        out string value)
+    {
+        if (!claims.TryGetValue(claimName, out var raw) || raw is null || string.IsNullOrWhiteSpace(raw))
```
Comment (jamesmblair) [inline src/SEBT.Portal.Infrastructure/Services/OidcVerificationClaimTranslator.cs:112]: Non-blocking nit: `string.IsNullOrWhiteSpace` is doing a null check first, so we don't need our own null check before that.

File: src/SEBT.Portal.Infrastructure/Services/OidcVerificationClaimTranslator.cs [hunk 2]
```diff
@@ -52,6 +62,63 @@ public OidcVerificationClaimTranslator(
...
+            ? DefaultFallbackDateClaimName
+            : _claimSettings.FallbackDateClaimName.Trim();
+
+    private static bool ClaimNamesAreEquivalent(string a, string b) =>
+        string.Equals(a, b, StringComparison.OrdinalIgnoreCase);
+
+    private static bool TryGetNonEmptyClaimValue(
+        IReadOnlyDictionary<string, string> claims,
+        string claimName,
+        out string value)
```
Comment (jamesmblair) [inline src/SEBT.Portal.Infrastructure/Services/OidcVerificationClaimTranslator.cs:110]: Non-blocking nit: `value` could be type `string?` and use a data annotation to indicate to the compiler it is only non-null if the method returns `true`

**Review by jamesmblair — APPROVED** [review]
Approved with a couple of non-blocking comments
